arch/platform/openmote/dev/antenna.c

```diff
@@ -52,25 +58,27 @@ void
 antenna_init(void)
 {
   /* Configure the ANT1 and ANT2 GPIO as output */
+  GPIO_SOFTWARE_CONTROL(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT);
+  GPIO_SOFTWARE_CONTROL(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT);
   GPIO_SET_OUTPUT(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT);
   GPIO_SET_OUTPUT(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT);
-
-  /* Select external antenna by default. */
-  antenna_external();
-}
-/*---------------------------------------------------------------------------*/
-void
-antenna_external(void)
-{
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT, 0);
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT, 1);
 }
 /*---------------------------------------------------------------------------*/
 void
-antenna_internal(void)
+antenna_select(uint8_t sel)
 {
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT, 0);
-  GPIO_WRITE_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT, 1);
+  if(sel == 0){
+    /* On OpenMote-CC2538, select external antenna
+     * On OpenMote-B     , select AT86RF215 */
+    GPIO_CLR_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_INT);
+    GPIO_SET_PIN(ANTENNA_BSP_RADIO_BASE, ANTENNA_BSP_RADIO_EXT);
+  }
```

Comment:
Code formatting: `} else {`

If you have `uncrustify` installed and run `tools/code-style/uncrustify-check-style.sh file.c`, you will get a diff for lines that do not follow the code style. Don't worry about lines you haven't changed.